experimental/host/mt7688/gpio.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package mt7688
+
+import (
+	"periph.io/x/periph/conn/pin"
+)
+
+// All the pins supported by the CPU.
+var (
+	GPIO0  *Pin // I2S_SDI
+	GPIO1  *Pin // I2S_SDO
+	GPIO2  *Pin // I2S_WS
+	GPIO3  *Pin // I2S_CLK
+	GPIO4  *Pin // I2C_SCLK
+	GPIO5  *Pin // I2C_SD
+	GPIO6  *Pin // SPI_CS1
+	GPIO7  *Pin // SPI_CLK
+	GPIO8  *Pin // SPI_MOSI
+	GPIO9  *Pin // SPI_MISO
+	GPIO10 *Pin // SPI_CS0
+	GPIO11 *Pin // GPIO0
+	GPIO12 *Pin // UART_TXD0
+	GPIO13 *Pin // UART_RXD0
+	GPIO14 *Pin // MDI_TP_P1
+	GPIO15 *Pin // MDI_TN_P1
+	GPIO16 *Pin // MDI_RP_P1
+	GPIO17 *Pin // MDI_RN_P1
+	GPIO18 *Pin // MDI_RP_P2
+	GPIO19 *Pin // MDI_RN_P2
+	GPIO20 *Pin // MDI_TP_P2
+	GPIO21 *Pin // MDI_TN_P2
+	GPIO22 *Pin // MDI_TP_P3
+	GPIO23 *Pin // MDI_TN_P3
+	GPIO24 *Pin // MDI_RP_P3
+	GPIO25 *Pin // MDI_RN_P3
+	GPIO26 *Pin // MDI_RP_P4
+	GPIO27 *Pin // MDI_RN_P4
+	GPIO28 *Pin // MDI_TP_P4
+	GPIO29 *Pin // MDI_TN_P4
+	GPIO30 *Pin // EPHY_LED4_N_JTRST_N (7688KN)
+	GPIO31 *Pin // EPHY_LED3_N_JTCLK (7688KN)
+	GPIO32 *Pin // EPHY_LED2_N_JTMS (7688KN)
+	GPIO33 *Pin // EPHY_LED1_N_JTDI (7688KN)
+	GPIO34 *Pin // EPHY_LED0_N_JTDO (7688KN)
+	GPIO35 *Pin // WLED_N (7688KN)
+	GPIO36 *Pin // PERST_N
+	GPIO37 *Pin // REF_CLKO
+	GPIO38 *Pin // WDT_RST_N
+	GPIO39 *Pin // EPHY_LED4_N_JTRST_N (7688AN)
+	GPIO40 *Pin // EPHY_LED3_N_JTCLK (7688AN)
+	GPIO41 *Pin // EPHY_LED2_N_JTMS (7688AN)
+	GPIO42 *Pin // EPHY_LED1_N_JTDI (7688AN)
+	GPIO43 *Pin // EPHY_LED0_N_JTDO (7688AN)
+	GPIO44 *Pin // WLED_N (7688AN)
+	GPIO45 *Pin // UART_TXD1
+	GPIO46 *Pin // UART_RXD1
+)
+
+// mappingMT7688 describes the mapping of the MT7688 processor gpios to their
+// alternate functions.
+//
+// It omits the in & out functions which are available on all gpio.
+//
+// The mapping is a combination of the naming from datasheet pages 25-31 and
+// the GPIO Pin Function Mapping on page 108:
+// https://labs.mediatek.com/fileMedia/download/9ef51e98-49b1-489a-b27e-391bac9f7bf3
+var mappingMT7688 = map[string][3]pin.Func{
```

Comment:
Another follow up is to convert this to a slice instead of a map; since the GPIOs have same numbering like for bcm283x, you can use a slice and the index correspond to the GPIO number.